Sources/TuistCore/Graph/GraphLoader.swift

```diff
@@ -105,17 +103,20 @@ public final class GraphLoader: GraphLoading {
         }
 
         cache.add(target: target, path: path)
-        let dependencies = try target.dependencies.compactMap { dependency in
-            try loadDependency(
+        let targetDependency = GraphDependency.target(name: name, path: path)
+        let dependencies: [GraphDependency] = try target.dependencies.compactMap { dependency in
+            guard let graphDep = try loadDependency(
                 path: path,
                 forPlatforms: target.supportedPlatforms,
                 dependency: dependency,
                 cache: cache
-            )
+            ) else { return nil }
+            cache.dependencyPlatformFilters[(targetDependency, graphDep)] = dependency.platformFilters
```

Comment:
Given we don't expect this to be heavily used (there may even be graphs that don't specify any filters at all!) we can only store explicitly specified filters as opposed to all filters for all edges. 

This means we can add some very easy initial optimisations such as skipping the platform filter traversals all together if there are no filters specified at all.